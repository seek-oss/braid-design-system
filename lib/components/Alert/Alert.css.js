export default {
  '.root': {
    display: 'flex',
  },
  '.icon': {
    display: 'flex',
    flexShrink: 0,
    flexGrow: 0,
  },
};
