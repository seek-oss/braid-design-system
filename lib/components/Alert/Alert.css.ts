import { style } from '@vanilla-extract/css';

export const toneBorder = style({
  opacity: 0.3,
});

export const cautionBorder = style({
  opacity: 0.6,
});
