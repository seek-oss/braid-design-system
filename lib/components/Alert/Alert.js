import React from 'react';
import PropTypes from 'prop-types';
import Box from '../Box/Box';
import Text from '../Text/Text';
import InfoIcon from '../icons/InfoIcon/InfoIcon';
import ErrorIcon from '../icons/ErrorIcon/ErrorIcon';
import styles from './Alert.css.js';

const iconForTone = tone => {
  if (tone === 'info') {
    return <InfoIcon fill="white" marginRight="small" />;
  }

  if (tone === 'critical') {
    return <ErrorIcon fill="critical" marginRight="small" />;
  }

  return null;
};

const textColorForTone = tone => {
  if (tone === 'info') {
    return 'white';
  }

  if (tone === 'critical') {
    return 'critical';
  }

  return null;
};

export default class Alert extends React.Component {
  static displayName = 'Alert';

  static propTypes = {
    tone: PropTypes.oneOf(['info', 'critical']).isRequired,
    children: PropTypes.node.isRequired
  };

  static defaultProps = {
    tone: 'info'
  };

  render() {
    const { tone, children, ...restProps } = this.props;

    const icon = iconForTone(tone);
    const color = textColorForTone(tone);

    return (
      <Box
        backgroundColor={tone}
        paddingLeft="gutter"
        paddingRight="gutter"
        paddingTop="medium"
        paddingBottom="medium"
        {...restProps}
      >
        <div className={styles.root}>
          <div className={styles.icon}>{icon}</div>
          <Text color={color} baseline={false}>
            {children}
          </Text>
        </div>
      </Box>
    );
  }
}
