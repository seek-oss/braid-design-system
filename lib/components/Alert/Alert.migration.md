# Alert Migration Guide

## API Changes

- The `message` prop has been removed in favour of passing `children`.
- `level={'primary' | 'secondary'}` have been deprecated in favour of a single visual weight. `level="tertiary"` has been replaced by the `Notice` component.
- No longer accepts arbitrary DOM properties, e.g. `className`. Please check that everything you need is exposed via the [public API.](https://seek-oss.github.io/braid-design-system/components/Alert)

## Todo List

- `hideIcon={true}`

## TBD

- `tone="help"`
- `pullout={boolean}`

## Diff

#### Primary/Secondary

```diff
 <Alert tone="positive"
-  level="primary"
-  message="Successfully completed"
-/>
+>
+  Successfully completed
+</Alert>
```

#### Tertiary

```diff
-<Alert
+<Notice
   tone="positive"
-  level="tertiary"
-  message="Successfully completed"
-/>
+>
+  Successfully completed
+</Notice>
```

## Previous Implementations

- [SEEK Style Guide](https://seek-oss.github.io/seek-style-guide/alert)
- [SEEK Asia Style Guide](https://seekinternational.github.io/seek-asia-style-guide/alert)
