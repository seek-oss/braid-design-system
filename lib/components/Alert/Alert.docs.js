import React from 'react';
import Alert from './Alert';

export default {
  component: Alert,
  examples: [
    {
      label: 'Info Alert',
      render: () => (
        <Alert tone="info">This is an important piece of information.</Alert>
      )
    },
    {
      label: 'Critical Alert',
      render: () => (
        <Alert tone="critical">This is a critical piece of information.</Alert>
      )
    }
  ]
};
