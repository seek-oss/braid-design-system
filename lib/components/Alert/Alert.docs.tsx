import React from 'react';
import { ComponentDocs } from '../../../site/src/types';
import { Alert } from './Alert';

const docs: ComponentDocs = {
  category: 'Content',
  migrationGuide: true,
  screenshotWidths: [320],
  examples: [
    {
      label: 'Info Alert',
      Example: () => (
        <Alert tone="info">This is an important piece of information.</Alert>
      ),
    },
    {
      label: 'Strong Info Alert',
      Example: () => (
        <Alert weight="strong" tone="info">
          This is an important piece of information.
        </Alert>
      ),
    },
    {
      label: 'Promote Alert',
      Example: () => (
        <Alert tone="promote">This is a promoted piece of information.</Alert>
      ),
    },
    {
      label: 'Strong Promote Alert',
      Example: () => (
        <Alert weight="strong" tone="promote">
          This is a promoted piece of information.
        </Alert>
      ),
    },
    {
      label: 'Critical Alert',
      Example: () => (
        <Alert tone="critical">This is a critical piece of information.</Alert>
      ),
    },
    {
      label: 'Strong Critical Alert',
      Example: () => (
        <Alert weight="strong" tone="critical">
          This is a critical piece of information.
        </Alert>
      ),
    },
    {
      label: 'Positive Alert',
      Example: () => (
        <Alert tone="positive">This is a positive piece of information.</Alert>
      ),
    },
    {
      label: 'Strong Positive Alert',
      Example: () => (
        <Alert weight="strong" tone="positive">
          This is a positive piece of information.
        </Alert>
      ),
    },
  ],
};

export default docs;
