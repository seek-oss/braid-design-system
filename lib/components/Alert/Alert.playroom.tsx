import React from 'react';
import { Alert as BraidAlert, AlertProps } from './Alert';

export const Alert = ({ tone, ...restProps }: AlertProps) => (
  <BraidAlert
    tone={typeof tone !== 'boolean' ? tone : undefined}
    {...restProps}
  />
);
