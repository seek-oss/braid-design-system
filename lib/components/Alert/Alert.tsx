import React, { ReactNode } from 'react';
import { useStyles } from 'sku/react-treat';
import { Box } from '../Box/Box';
import { Text } from '../Text/Text';
import { InfoIcon } from '../icons/InfoIcon/InfoIcon';
import { ErrorIcon } from '../icons/ErrorIcon/ErrorIcon';
import { TickCircleIcon } from '../icons/TickCircleIcon/TickCircleIcon';
import * as styleRefs from './Alert.treat';

type Tone = 'info' | 'critical' | 'positive';
type AlertWeight = 'strong' | 'regular';

export interface AlertProps {
  tone?: Tone;
  weight?: AlertWeight;
  children: ReactNode;
}

const backgroundColorForTone = (tone: Tone, weight: AlertWeight) => {
  if (weight === 'strong') {
    return tone;
  }

  if (tone === 'positive') {
    return 'positiveLight';
  }

  if (tone === 'critical') {
    return 'criticalLight';
  }

  if (tone === 'info') {
    return 'infoLight';
  }
};

const icons = {
  info: InfoIcon,
  critical: ErrorIcon,
  positive: TickCircleIcon,
};

export const Alert = ({
  tone = 'info',
  weight = 'regular',
  children,
}: AlertProps) => {
  const styles = useStyles(styleRefs);
  const backgroundColor = backgroundColorForTone(tone, weight);
  const color = weight === 'strong' ? 'white' : tone;
  const Icon = icons[tone];

  return (
    <Box
      backgroundColor={backgroundColor}
      paddingLeft="gutter"
      paddingRight="gutter"
      paddingTop="medium"
      paddingBottom="medium"
      display="flex"
    >
      {Icon ? (
        <Box paddingRight="small" display="flex" className={styles.icon}>
          <Icon fill={color} />
        </Box>
      ) : null}
      <Text color={color} baseline={false}>
        {children}
      </Text>
    </Box>
  );
};
