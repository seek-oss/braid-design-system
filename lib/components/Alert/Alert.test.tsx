import React from 'react';
import { Alert } from '../';
import { AlertProps } from './Alert';
import { render } from 'react-testing-library';
import forEachTheme from '../../test/utils/forEachTheme';

describe('Alert', () => {
  forEachTheme(({ ThemeProvider }) => {
    describe('Tones', () => {
      const tones: Array<AlertProps['tone']> = ['critical', 'info', undefined];

      tones.forEach(tone => {
        test(tone || 'default', () => {
          const { container } = render(
            <ThemeProvider>
              <Alert tone={tone}>{tone}</Alert>
            </ThemeProvider>
          );
          expect(container.firstChild).toMatchSnapshot();
        });
      });
    });
  });
});
