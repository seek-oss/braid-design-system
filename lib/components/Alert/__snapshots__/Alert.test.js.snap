// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Theme: jobStreet Tones critical 1`] = `
<div
  class="Box__root backgroundColor__critical borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__critical Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 512 512"
        width="16"
      >
        <path
          d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__critical fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        critical
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: jobStreet Tones info 1`] = `
<div
  class="Box__root backgroundColor__info borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__white Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 30 30"
        width="16"
        x="0"
        y="0"
      >
        <path
          d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
        />
        <path
          d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
        />
        <path
          d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__white fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        info
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: seekAnz Tones critical 1`] = `
<div
  class="Box__root backgroundColor__critical borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__critical Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 512 512"
        width="16"
      >
        <path
          d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__critical fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        critical
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: seekAnz Tones info 1`] = `
<div
  class="Box__root backgroundColor__info borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__white Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 30 30"
        width="16"
        x="0"
        y="0"
      >
        <path
          d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
        />
        <path
          d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
        />
        <path
          d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__white fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        info
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: seekAsia Tones critical 1`] = `
<div
  class="Box__root backgroundColor__critical borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__critical Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 512 512"
        width="16"
      >
        <path
          d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__critical fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        critical
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: seekAsia Tones info 1`] = `
<div
  class="Box__root backgroundColor__info borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__white Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 30 30"
        width="16"
        x="0"
        y="0"
      >
        <path
          d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
        />
        <path
          d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
        />
        <path
          d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__white fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        info
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: wireframe Tones critical 1`] = `
<div
  class="Box__root backgroundColor__critical borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__critical Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 512 512"
        width="16"
      >
        <path
          d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__critical fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        critical
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert Theme: wireframe Tones info 1`] = `
<div
  class="Box__root backgroundColor__info borderColor__standard paddingTop__medium paddingRight__gutter paddingBottom__medium paddingLeft__gutter"
>
  <div
    class="Alert__root"
  >
    <div
      class="Alert__icon"
    >
      <svg
        class="width__standardText height__standardText fill__white Box__root borderColor__standard marginRight__small"
        height="16"
        viewBox="0 0 30 30"
        width="16"
        x="0"
        y="0"
      >
        <path
          d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
        />
        <path
          d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
        />
        <path
          d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
        />
      </svg>
    </div>
    <div
      class="Text__block Box__root borderColor__standard"
    >
      <span
        class="Text__block color__white fontFamily__text fontSize__standard fontWeight__regular"
        style="transform: none;"
      >
        info
      </span>
    </div>
  </div>
</div>
`;
